tests/unit/test_global_config.py

```diff
@@ -35,6 +35,23 @@ def test_global_config_parse(filename):
     assert global_config.has_dashboard() is True
 
 
+def test_global_config_equal():
+    global_config_1 = global_config_2 = global_config_lib.GlobalConfig.from_file(
+        helpers.get_testdata_file_path("valid_config.json")
+    )
+
+    assert global_config_1 == global_config_2
```

Comment:
Could we also add the below:
```
    assert global_config_1 == global_config_2
    assert global_config_2 == global_config_1
    assert global_config_1 == global_config_2
    assert global_config_2 == global_config_1
    assert not global_config_2 != global_config_1
```
I know it looks redundant and vague, but it ensures the functionality of `__eq__()` implementation in all scenarios, if ever some breakage was introduced by Python's patch or minor releases.

Reply:
I am not sure we need to test all
those alterations in our code. I do not want to double test what Python itself should be testing. They should be supporting this API and I think it's pretty safe to leave it as it is.

Reply:
Yes, the only reason being I had seen such issues w.r.t. to eq and hash implementations. We can skip them, in that case.